### PR TITLE
fix: added clear search on removing badge and added keyboard shortcuts

### LINKED ISSUE #9401: Improve Facility & Phone number Search Field Reset and Dynamic Shortcut Hints for new search component
**Describe the bug**

**Issue 1:**
When closing the badge related to the facility search, while the badge is removed and the applied filter is reset on the page, the input data in the search field is not being cleared as expected.

Similar issue is being experience for phone number search in the patient tab

**Issue 2:**
The newly refactored search component is missing the following keyboard shortcuts, which were available in the previous version:

Ctrl+K: Shortcut to focus on the search field.
Esc: Shortcut to clear the search field.
For reference to the old component behavior, visit the previous deployment: [Old Deployment](https://f5958847.care-fe.pages.dev/facility).

**To Reproduce**

1. Go to facility tab
2. search for a facility
3. close the related badge

**Expected behavior**

- [ ] Reset Input on Badge Removal: Closing the badge associated with a facility search should also reset the input field in the search box to ensure consistency in the UI.
- [ ] Reintroduce Keyboard Shortcuts: Ctrl+K for Focusing the Field:
- The shortcut should focus the cursor on the search field.
- The field should display a Ctrl+K hint when it is empty, to inform users of the shortcut functionality.
- [ ] Esc for Clearing the Field:
- If the search field contains data, pressing Esc should clear the input.
- The field should display an Esc hint when it is not empty, indicating that the user can use this shortcut to reset it.
- [ ] Similar behavior should be implemented for the User Tab search field, where the shortcut hints are dynamic. Currently, only Ctrl+K is displayed statically.

**Screenshots**

Image
Image

### PR DESCRIPTION
**Pull request by noufalrahim**
## Proposed Changes

- Fixes #9401 
- Added keyboard shortcuts to clear and search
- Added clear search if badge is closed
- More?

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
	- Enhanced search component to reset input when the "clearSearch" parameter includes "search".
	- Added keyboard shortcuts: clear input with "Escape" and focus input with "Ctrl+K" or "Meta+K".

- **Bug Fixes**
	- Improved error handling by resetting error state upon option selection.

### CODE REVIEW

File: src/components/Common/SearchByMultipleFields.tsx [hunk 1]
```diff
@@ -84,6 +84,9 @@ const SearchByMultipleFields: React.FC<SearchByMultipleFieldsProps> = ({
         .some((element) => clearSearch.params?.includes(element));
       clearinput ? setSearchValue("") : null;
       inputRef.current?.focus();
+      if (clearSearch?.params && clearSearch?.params.includes("search")) {
+        setSearchValue("");
+      }
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Common/SearchByMultipleFields.tsx]: Try this implementation @noufalrahim

**Comment by noufalrahim** [comment]
@Jacobjeevan is this ok..?

**Comment by noufalrahim** [comment]
@nihal467 @rithviknishad can u check this pr?

**Comment by bodhish** [comment]
@noufalrahim please complete the Merge Checklist.

File: src/components/Common/SearchByMultipleFields.tsx [hunk 2]
```diff
@@ -110,6 +114,14 @@ const SearchByMultipleFields: React.FC<SearchByMultipleFieldsProps> = ({
         e.preventDefault();
         setOpen(true);
       }
+      if ((e.ctrlKey || e.metaKey) && e.key === "k") {
```
Comment (Jacobjeevan) [inline src/components/Common/SearchByMultipleFields.tsx:117]: LGTM so far, but check the expected behavior checklist in the issue. There's a few items missing (displaying the shortcut hint).

**Review by Jacobjeevan — CHANGES_REQUESTED** [review]

**Comment by rithviknishad** [comment]
no longer relevant in new architecture

**Pull request closed without merging**